test/setup/loadLocalDockerImage.ts

```diff
@@ -0,0 +1,36 @@
+import { SpawnSyncReturns, spawnSync } from 'child_process';
+
+// https://regex101.com/r/eK9lPd/3
+// eslint-disable-next-line security/detect-unsafe-regex
+export const dockerImageUrlRegExp = /^(?<Name>(?<=^)(?:(?<Domain>(?:(?:localhost|[\w-]+(?:\.[\w-]+)+)(?::\d+)?)|[\w]+:\d+)\/)?\/?(?<Namespace>(?:(?:[a-z0-9]+(?:(?:[._]|__|[-]*)[a-z0-9]+)*)\/)*)(?<Repo>[a-z0-9-]+))[:@]?(?<Reference>(?<=:)(?<Tag>[\w][\w.-]{0,127})|(?<=@)(?<Digest>[A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][0-9A-Fa-f]{32,}))?/;
```

Comment:
This is a transcription of Docker's own URL parsing RegEx. For our purposes this could be done in a more naive way, but I think this is going to serve us better. The regex101.com link in the comment tracks changes and any issues, as well as it gives a sandbox for testing the RegEx.